File: UtilisateurService/src/main/java/org/ai/utilisateurservice/Service/UtilisateurService.java
package org.ai.utilisateurservice.Service;

import org.ai.utilisateurservice.Entity.Utilisateur;

import java.util.List;

public interface UtilisateurService {
    Utilisateur createUser(Utilisateur utilisateur);
    Utilisateur updateUser(Long id, Utilisateur utilisateur);
    Utilisateur getUserById(Long id);
    List<Utilisateur> getAllUsers();
    List<Utilisateur> obtenirUtilisateursParRole(String role);
    void deleteUser(Long id);
}

File: e-commerce/src/app/services/utilisateur.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Utilisateur } from '../models/utilisateur.model';

@Injectable({
  providedIn: 'root'
})
export class UtilisateurService {
  private apiUrl = 'http://localhost:8080/api/utilisateurs';

  constructor(private http: HttpClient) {}

  getAllUsers(): Observable<Utilisateur[]> {
    return this.http.get<Utilisateur[]>(this.apiUrl);
  }

  getUserById(id: number): Observable<Utilisateur> {
    return this.http.get<Utilisateur>(`${this.apiUrl}/${id}`);
  }

  createUser(utilisateur: Utilisateur): Observable<Utilisateur> {
    return this.http.post<Utilisateur>(this.apiUrl, utilisateur);
  }

  updateUser(id: number, utilisateur: Utilisateur): Observable<Utilisateur> {
    return this.http.put<Utilisateur>(`${this.apiUrl}/${id}`, utilisateur);
  }

  deleteUser(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
